x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorCustomSchedule.java

```diff
@@ -0,0 +1,292 @@
+            .setName((String) args[4])
+            .build()
+    );
+
+    static {
+        PARSER.declareField(
+            constructorArg(),
+            (parser, context) -> parser.map(HashMap::new, ConfigurationOverride::fromXContent),
+            CONFIG_OVERRIDES_FIELD,
+            ObjectParser.ValueType.OBJECT
+        );
+        PARSER.declareBoolean(constructorArg(), ENABLED_FIELD);
+        PARSER.declareString(constructorArg(), INTERVAL_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_SYNCED_FIELD);
+        PARSER.declareString(constructorArg(), NAME_FIELD);
+    }
+
+    public static ConnectorCustomSchedule fromXContent(XContentParser parser) throws IOException {
+        return PARSER.parse(parser, null);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(CONFIG_OVERRIDES_FIELD.getPreferredName(), configurationOverrides);
+        builder.field(ENABLED_FIELD.getPreferredName(), enabled);
+        builder.field(INTERVAL_FIELD.getPreferredName(), interval);
+
+        if (lastSynced != null) {
+            builder.field(LAST_SYNCED_FIELD.getPreferredName(), lastSynced);
+        }
+
+        builder.field(NAME_FIELD.getPreferredName(), name);
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(configurationOverrides, StreamOutput::writeString, StreamOutput::writeGenericValue);
+        out.writeBoolean(enabled);
+        out.writeString(interval);
+        out.writeString(lastSynced);
+        out.writeString(name);
+    }
+
+    public static class Builder {
+
+        private Map<String, ConfigurationOverride> configurationOverrides;
+        private boolean enabled;
+        private String interval;
+        private String lastSynced;
+        private String name;
+
+        public Builder setConfigurationOverrides(Map<String, ConfigurationOverride> configurationOverrides) {
+            this.configurationOverrides = configurationOverrides;
+            return this;
+        }
+
+        public Builder setEnabled(boolean enabled) {
+            this.enabled = enabled;
+            return this;
+        }
+
+        public Builder setInterval(String interval) {
+            this.interval = interval;
+            return this;
+        }
+
+        public Builder setLastSynced(String lastSynced) {
+            this.lastSynced = lastSynced;
+            return this;
+        }
+
+        public Builder setName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public ConnectorCustomSchedule build() {
+            return new ConnectorCustomSchedule(configurationOverrides, enabled, interval, lastSynced, name);
+        }
+    }
+
+    public static class ConfigurationOverride implements Writeable, ToXContentObject {
```

Comment:
Is this a placeholder definition for `ConfigurationOverride`? It looks to be specific to the crawler connector.

Reply:
This is a definition (not a placeholder IMO) of an allowed configuration override. This is only used by crawler (and not connectors). In the past there was some effort to unify representation of a generic connector and crawler in a single connector doc - hence it's a bit mixed up right now as we are in an in-between state. 

Here is the link to details: https://www.elastic.co/guide/en/enterprise-search/8.9/crawler-managing.html#crawler-managing-multiple-schedules

Reply:
Got it, thanks for the background!